<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Simon Glass</author>
 <comment>Dialogue to allow scanning of new paper</comment>
 <class>Pscan</class>
 <widget class="QDialog" name="Pscan">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>521</width>
    <height>312</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Page Scanning</string>
  </property>
  <property name="windowIconText">
   <string> df</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_3">
   <item>
    <layout class="QVBoxLayout">
     <item>
      <layout class="QHBoxLayout">
       <item>
        <widget class="QLabel" name="textLabel1">
         <property name="text">
          <string>Stack name</string>
         </property>
         <property name="wordWrap">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Minimum</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLineEdit" name="stackName">
         <property name="toolTip">
          <string>Enter the name of the stack and press 'scan' to scan it</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout">
       <item>
        <widget class="QLabel" name="textLabel2">
         <property name="text">
          <string>Page name</string>
         </property>
         <property name="wordWrap">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Minimum</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>10</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLineEdit" name="pageName">
         <property name="toolTip">
          <string>Enter the name for each page of the stack to be scanned</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QLabel" name="testLabel">
         <property name="text">
          <string>Folder</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Minimum</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>10</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLineEdit" name="folderName">
         <property name="toolTip">
          <string>Type here to interactively search for folders</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <spacer>
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeType">
        <enum>QSizePolicy::Expanding</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>21</width>
         <height>95</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <layout class="QHBoxLayout">
       <item>
        <layout class="QVBoxLayout">
         <item>
          <widget class="QPushButton" name="options">
           <property name="toolTip">
            <string>Change scan options</string>
           </property>
           <property name="text">
            <string>Options</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="config">
           <property name="toolTip">
            <string>Change maxview options</string>
           </property>
           <property name="text">
            <string>Config</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer>
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Expanding</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>41</width>
             <height>71</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout">
         <item>
          <widget class="QPushButton" name="reset">
           <property name="toolTip">
            <string>Reset to default settings</string>
           </property>
           <property name="text">
            <string>Reset</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="source">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="toolTip">
            <string>Select a different scanner to use</string>
           </property>
           <property name="text">
            <string>Select scanner...</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="settings">
           <property name="toolTip">
            <string>Control all scanner features explicitly</string>
           </property>
           <property name="text">
            <string>Advanced...</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout">
       <item>
        <layout class="QHBoxLayout">
         <item>
          <spacer>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Maximum</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="infoStr">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>60</width>
             <height>0</height>
            </size>
           </property>
           <property name="toolTip">
            <string>Displays page coverage information</string>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
           <property name="wordWrap">
            <bool>false</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout">
         <item>
          <widget class="QLabel" name="progressStr">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="toolTip">
            <string>Displays page and stack number while scanning</string>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="wordWrap">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <spacer>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Maximum</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="sizeStr">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>60</width>
             <height>0</height>
            </size>
           </property>
           <property name="toolTip">
            <string>Displays total stack size in bytes</string>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="wordWrap">
            <bool>false</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QProgressBar" name="progressBar">
       <property name="toolTip">
        <string>Displays scanning progress</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QPushButton" name="cancel">
         <property name="toolTip">
          <string>Cancel the current scan</string>
         </property>
         <property name="text">
          <string>Cancel</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="stop">
         <property name="toolTip">
          <string>Stop the current scan after this page has been scanned</string>
         </property>
         <property name="text">
          <string>Stop</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="scan">
         <property name="toolTip">
          <string>Perform a scan</string>
         </property>
         <property name="text">
          <string>Scan</string>
         </property>
         <property name="default">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QComboBox" name="preset"/>
     </item>
     <item>
      <layout class="QGridLayout" name="gridLayout">
       <item row="1" column="0">
        <widget class="QRadioButton" name="grey">
         <property name="toolTip">
          <string>Select greyscale scanning</string>
         </property>
         <property name="text">
          <string>Grey</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">format</string>
         </attribute>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QRadioButton" name="mono">
         <property name="toolTip">
          <string>Select monochrome scanning (thresholded blank and white)</string>
         </property>
         <property name="text">
          <string>Mono</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">format</string>
         </attribute>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QRadioButton" name="dither">
         <property name="toolTip">
          <string>Select monochrome scanning with dithering</string>
         </property>
         <property name="text">
          <string>Dither</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">format</string>
         </attribute>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QRadioButton" name="colour">
         <property name="toolTip">
          <string>Select colour scanning</string>
         </property>
         <property name="text">
          <string>Colour</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">format</string>
         </attribute>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout">
       <item>
        <layout class="QHBoxLayout">
         <item>
          <widget class="QCheckBox" name="adf">
           <property name="toolTip">
            <string>Feed from the automatic document feeder</string>
           </property>
           <property name="text">
            <string>ADF</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="duplex">
           <property name="toolTip">
            <string>Scan both sides of the paper</string>
           </property>
           <property name="text">
            <string>Duplex</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout">
         <item>
          <widget class="QLabel" name="textLabel3">
           <property name="text">
            <string>DPI</string>
           </property>
           <property name="wordWrap">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="res">
           <property name="toolTip">
            <string>Select the dots per inch (300 gives a good scan, 200 is fax quality)</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout">
         <item>
          <widget class="QLabel" name="textLabel3_2">
           <property name="text">
            <string>Size</string>
           </property>
           <property name="wordWrap">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="pageSize">
           <property name="minimumSize">
            <size>
             <width>150</width>
             <height>0</height>
            </size>
           </property>
           <property name="toolTip">
            <string>Select paper size</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="SliderSpin" name="bright" native="true">
         <property name="minimumSize">
          <size>
           <width>210</width>
           <height>30</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Select brightness or threshold</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="SliderSpin" name="contrast" native="true">
         <property name="minimumSize">
          <size>
           <width>210</width>
           <height>30</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Select contrast</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Expanding</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>1</width>
           <height>1</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="2" margin="2"/>
 <customwidgets>
  <customwidget>
   <class>SliderSpin</class>
   <extends>QWidget</extends>
   <header>sliderspin.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>stackName</tabstop>
  <tabstop>pageName</tabstop>
  <tabstop>mono</tabstop>
  <tabstop>dither</tabstop>
  <tabstop>grey</tabstop>
  <tabstop>colour</tabstop>
  <tabstop>adf</tabstop>
  <tabstop>duplex</tabstop>
  <tabstop>res</tabstop>
  <tabstop>pageSize</tabstop>
  <tabstop>cancel</tabstop>
  <tabstop>stop</tabstop>
  <tabstop>scan</tabstop>
  <tabstop>options</tabstop>
  <tabstop>reset</tabstop>
  <tabstop>config</tabstop>
  <tabstop>source</tabstop>
  <tabstop>settings</tabstop>
 </tabstops>
 <includes>
  <include location="local">mainwidget.h</include>
  <include location="local">qscanner.h</include>
  <include location="local">qscandialog.h</include>
  <include location="local">sliderspin.h</include>
  <include location="local">sliderspin.h</include>
 </includes>
 <resources/>
 <connections>
  <connection>
   <sender>bright</sender>
   <signal>signalValueChanged(int)</signal>
   <receiver>Pscan</receiver>
   <slot>brightChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pageSize</sender>
   <signal>activated(int)</signal>
   <receiver>Pscan</receiver>
   <slot>size_activated(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>res</sender>
   <signal>activated(int)</signal>
   <receiver>Pscan</receiver>
   <slot>res_activated(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>duplex</sender>
   <signal>clicked()</signal>
   <receiver>Pscan</receiver>
   <slot>duplex_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>adf</sender>
   <signal>clicked()</signal>
   <receiver>Pscan</receiver>
   <slot>adf_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>scan</sender>
   <signal>clicked()</signal>
   <receiver>Pscan</receiver>
   <slot>scan_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>settings</sender>
   <signal>clicked()</signal>
   <receiver>Pscan</receiver>
   <slot>settings_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>source</sender>
   <signal>clicked()</signal>
   <receiver>Pscan</receiver>
   <slot>source_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>reset</sender>
   <signal>clicked()</signal>
   <receiver>Pscan</receiver>
   <slot>reset_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>contrast</sender>
   <signal>signalValueChanged(int)</signal>
   <receiver>Pscan</receiver>
   <slot>contrastChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>options</sender>
   <signal>clicked()</signal>
   <receiver>Pscan</receiver>
   <slot>options_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <buttongroups>
  <buttongroup name="format"/>
 </buttongroups>
</ui>
